modules/machineset-yaml-alibaba.adoc

```diff
@@ -0,0 +1,176 @@
+            id: <resource_group_id> <8>
+            type: ID
+          securityGroups:
+          - tags: <9>
+            - Key: kubernetes.io/cluster/<infrastructure_id>
+              Value: owned
+            - Key: GISV
+              Value: ocp
+            - Key: sigs.k8s.io/cloud-provider-alibaba/origin
+              Value: ocp
+            - Key: Name
+              Value: <infrastructure_id>-sg-<role>
+            type: Tags
+          systemDisk: <10>
+            category: cloud_essd
+            size: <disk_size>
+          tag: <9>
+          - Key: kubernetes.io/cluster/<infrastructure_id>
+            Value: owned
+          - Key: GISV
+            Value: ocp
+          - Key: sigs.k8s.io/cloud-provider-alibaba/origin
+            Value: ocp
+          userDataSecret:
+            name: <user_data_secret> <11>
+          vSwitch:
+            tags: <9>
+            - Key: kubernetes.io/cluster/<infrastructure_id>
+              Value: owned
+            - Key: GISV
+              Value: ocp
+            - Key: sigs.k8s.io/cloud-provider-alibaba/origin
+              Value: ocp
+            - Key: Name
+              Value: <infrastructure_id>-vswitch-<region>-<zone>
+            type: Tags
+          vpcId: ""
+          zoneId: <region>-<zone> <12>
+ifdef::infra[]
+        taints: <13>
+        - key: node-role.kubernetes.io/infra
+          effect: NoSchedule
+endif::infra[]
+----
+<1> Specify the infrastructure ID that is based on the cluster ID that you set when you provisioned the cluster. If you have the OpenShift CLI installed, you can obtain the infrastructure ID by running the following command:
++
+[source,terminal]
+----
+$ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
+----
+ifndef::infra[]
+<2> Specify the node label to add.
+<3> Specify the infrastructure ID, node label, and region.
+endif::infra[]
+ifdef::infra[]
+<2> Specify the `<infra>` node label.
+<3> Specify the infrastructure ID, `<infra>` node label, and region.
+endif::infra[]
+<4> Specify the image to use. Note: Get an image from an existing default machine set for the cluster.
+<5> Specify the instance type you want to use for the machine set.
+<6> Specify the name of the RAM role to use for the machine set .
+// Should they use the `ramRoleName` value that the installer populates in the default machine set?
+<7> Specify the region to place machines on.
+<8> Specify the resource group for the cluster. You can use the value in the machine set that that the installer creates, or specify a different one.
+<9> Specify the tags to use for the machine set. Minimally, you must include the tags that the installer populates in the default machine set it creates. You can include additional tags as needed.
+// Would it be better to leave the information here more vague so that the user doesn't get hung up on specifics? For instance, replace `kubernetes.io/cluster/<infrastructure_id>` with `<tag_value_1>`?
```

Comment:
Question for devs.

Reply:
The tag [1] is mandatory, I am not sure if it will be added by the controller even it's not defined on the MachineSet template, I believe it will not.
So I would keep at least the tag[1] to avoid mistakes.

[1] cluster tag
```
            - name: kubernetes.io/cluster/<infrastructure_id> 
              value: owned
```

Reply:
ok, good to know. To follow up:
1. Are all of these mandatory?
```
          - tags: <9>
            - Key: kubernetes.io/cluster/<infrastructure_id>
              Value: owned
            - Key: GISV
              Value: ocp
            - Key: sigs.k8s.io/cloud-provider-alibaba/origin
              Value: ocp
            - Key: Name
              Value: <infrastructure_id>-sg-<role>
            type: Tags
```
2. I think all of those are in the default machine set the installer makes, but can you confirm that?
3. Is it true that they could also add more tags if they know what to add for some use case they have in mind?

Reply:
1. I am not sure. Let's get help from our Alibaba colleagues. As commented on Slack, I am able to create Machines through MachineSet **with only** one tag: `kubernetes.io/cluster/<infrastructure_id>`
```
$ ./oc --kubeconfig auth/kubeconfig \
  -n openshift-machine-api \
  get machineset mrbdoc-xkqw9-wk-custom-use1a -o json  |jq .spec.template.spec.providerSpec.value.tag
[
  {
    "Key": "kubernetes.io/cluster/mrbdoc-xkqw9",
    "Value": "owned"
  }
]

```
2. Yes, those tags are the default from the release I've tested: `4.10.0-rc.0` 
3. Yes, it is possible to add custom tags alongside the mandatory ones.
```
$ ./oc --kubeconfig auth/kubeconfig \
  -n openshift-machine-api \
  get machineset mrbdoc-xkqw9-wk02-use1a -o json |jq .spec.template.spec.providerSpec.value.tag
[
(...)
  {
    "Key": "myTag",
    "Value": "myValue"
  }
]

$ ./oc --kubeconfig auth/kubeconfig \
  -n openshift-machine-api
  get machine mrbdoc-xkqw9-wk02-use1a-glsl7 -o json |jq .spec.providerSpec.value.tag
[
(...)
  {
    "Key": "myTag",
    "Value": "myValue"
  }
]

$ aliyun ecs DescribeTags \
  --RegionId us-east-1 \
  --ResourceId ${instance_id} |jq -r '.Tags.Tag[] | select(.TagKey=="myTag")'
{
  "ResourceTypeCount": {
    "Instance": 1
  },
  "TagKey": "myTag",
  "TagValue": "myValue"
}

```